src/main/java/it/grid/storm/catalogs/timertasks/RequestsGarbageCollector.java

```diff
@@ -145,16 +145,19 @@ synchronized private int purgeExpiredRequests(int purgeSize,
 	}
 
 	/**
-	 * Method used to purge from db a bunch of completed recall requests.
+	 * Method used to clear a bunch of completed recall requests from database.
 	 * 
+	 * @param expirationTime
+	 *          The number of seconds that must pass before considering a 
+	 *          request as expired
 	 * @param purgeSize
 	 *          The maximum size of the bunch of expired requests that must be
 	 *          deleted
 	 * @return The number of requests involved.
 	 */
-	synchronized private int purgeExpiredRecallRequests(int purgeSize) {
+	private synchronized int purgeExpiredRecallRequests(long expirationTime, int purgeSize) {
 
-		int n = new TapeRecallCatalog().purgeCatalog(purgeSize);
+		int n = new TapeRecallCatalog().purgeCatalog(expirationTime, purgeSize);
 		if (n == 0) {
 			log.trace("No entries have been purged from tape_recall table");
```

Comment:
I would remove the if an always log the number of entries removed, which can be 0